### PR TITLE
Fix coverity scan errors from elf refactoring

### PR DESCRIPTION
**Pull request by ghost**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

Fix various coverity scan errors:

* 349004: removed some code
* 349003: add a check
* 349002, 348997: check the result of BREAD32 (there isn't any upper bound)
* 349001, 38999: free the string
* 349000, 38998: free the string
* **348996: I have no idea how to fix that because there isn't any upper bound**

**Test plan**

...

**Closing issues**

...

### CODE REVIEW

File: librz/bin/format/elf/elf_symbols.c [hunk 1]
```diff
@@ -216,6 +216,11 @@ static size_t get_number_of_symbols_from_gnu_hash(ELFOBJ *bin) {
 	ut32 number_of_bucket = BREAD32(bin->b, pos);
 	ut32 symbol_base = BREAD32(bin->b, pos);
 	ut32 bitmask_nwords = BREAD32(bin->b, pos);
+
+	if (number_of_bucket == UT32_MAX || symbol_base == UT32_MAX || bitmask_nwords == UT32_MAX) {
```
Comment (wargio) [inline librz/bin/format/elf/elf_symbols.c:220]: shouldn't this be compared with the file size instead with its max?

Comment (ghost) [reply librz/bin/format/elf/elf_symbols.c:220]: The error value is UT32_MAX.

Comment (ret2libc) [reply librz/bin/format/elf/elf_symbols.c:220]: In addition to checking the error you might want to check if they are "valid" values as well. I guess that's what @wargio means.

**Review by ret2libc — APPROVED** [review]

File: librz/bin/format/elf/elf_symbols.c (hunk 1 above)
Comment (wargio) [reply librz/bin/format/elf/elf_symbols.c:220]: what i mean is that i think you should check the plausibility of having for example `0xefefefef` as `number_of_bucket ` in an elf file of `32 kb` for example.

**Comment by wargio** [comment]
BTW, what is `348996` error ?

**Comment by ghost** [comment]
> BTW, what is `348996` error ?

image

File: librz/bin/format/elf/elf_symbols.c (hunk 1 above)
Comment (ghost) [reply librz/bin/format/elf/elf_symbols.c:220]: something like
```c
if (number_of_bucket * sizeof(ut32) + offset >= bin->size) {
	return 0
}
```

**Review by XVilka — APPROVED** [review]

File: librz/bin/format/elf/elf_symbols.c (hunk 1 above)
Comment (wargio) [reply librz/bin/format/elf/elf_symbols.c:220]: yes exactly

**Pull request merged**